tensorboard/plugins/interactive_inference/witwidget/notebook/colab/wit.py

```diff
@@ -12,4 +12,339 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# TODO(jameswex): Colab implementation will be in separate PR.
+import base64
+import json
+import tensorflow as tf
+from IPython import display
+from google.colab import output
+from google.protobuf import json_format
+from tensorboard.plugins.interactive_inference.utils import inference_utils
+
+
+# Python functions for requests from javascript.
+def infer_examples(wit_id):
+  WitWidget.widgets[wit_id].infer()
+output.register_callback('notebook.InferExamples', infer_examples)
+
+
+def delete_example(wit_id, index):
+  WitWidget.widgets[wit_id].delete_example(index)
+output.register_callback('notebook.DeleteExample', delete_example)
+
+
+def duplicate_example(wit_id, index):
+  WitWidget.widgets[wit_id].duplicate_example(index)
+output.register_callback('notebook.DuplicateExample', duplicate_example)
+
+
+def update_example(wit_id, index, example):
+  WitWidget.widgets[wit_id].update_example(index, example)
+output.register_callback('notebook.UpdateExample', update_example)
+
+
+def get_eligible_features(wit_id):
+  WitWidget.widgets[wit_id].get_eligible_features()
+output.register_callback('notebook.GetEligibleFeatures', get_eligible_features)
+
+
+def infer_mutants(wit_id, details):
+  WitWidget.widgets[wit_id].infer_mutants(details)
+output.register_callback('notebook.InferMutants', infer_mutants)
+
+
+# HTML/javascript for the WIT frontend.
+WIT_HTML = """
```

Comment:
It would be wonderful if this is created/went through compilation/Vulcanization and read as a string. Would it be possible?